File: tests/integration/construct_component_methods_unknown/dotnet/Random.cs
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

using Pulumi;

class RandomArgs : ResourceArgs
{
    [Input("length")]
    public Input<int> Length { get; set; } = null!;
}

class Random : CustomResource
{
    public Random(string name, RandomArgs args, CustomResourceOptions? opts = null)
        : base("testprovider:index:Random", name, args, opts)
    {
    }
}

File: tests/integration/construct_component_methods_unknown/nodejs/random.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

interface RandomArgs {
    length: pulumi.Input<number>;
}

export class Random extends pulumi.CustomResource {
    public readonly length!: pulumi.Output<number>;
    public readonly result!: pulumi.Output<string>;
    constructor(name: string, args: RandomArgs, opts?: pulumi.CustomResourceOptions) {
        super("testprovider:index:Random", name, args, opts);
    }
}
